### ISSUE TITLE
Build first version of data loader

### ISSUE THREAD

**Issue by raspstephan** [raspstephan/nwp-downscale#10]
Now that we have  some first data downloaded and preprocessed, the next step is to write a data generator so that we can feed the data to our NNs.

@anna-184702, maybe you have some insights here for how to do this in PyTorch. I probably would start with a PyTorch Dataset and DataLoader following https://pytorch.org/tutorials/beginner/data_loading_tutorial.html

For starters, we can attempt to just open the xarray datasets and do the computations on the fly. Eventually, I anticipate this being too slow or require too much RAM. Then we have to think about some way of speeding things up but let's discuss that once we get there.

**To do**
- [x] Include mask creation in Dataset
- [x] Normalization (Ended up doing a min-max scaling).
- [x] Precipitation log-transform
- [x] Add weighted sampling according to precipitation amount
- [x] Add constant fields to X (orography, land-sea mask)
- [x] Add other TIGGE fields to X
- [x] Option for wider TIGGE input view
- [x] Add train/valid split based on day of month
- [x] **Make sure that the patches and times are correct!!!** (Someone besides me should check this to make sure 😄)

**Comment by raspstephan** [comment]
I started building a first crude version. I got to the point where it reads TIGGE and MRMS data and returns patches. Lot's of things to do before it's usable for training. The data loader is in notebook 04 of the dataloader branch.

I added a to do list in the post above to keep track of all the required enhancements.

**Comment by raspstephan** [comment]
Updates #13 and #14 now implement a fully working dataloader that can be used to test some first networks!

**Comment by raspstephan** [comment]
After a sanity check by @HirtM, let's close this issue and treat enhancements as new issues.

**Comment by HirtM** [comment]
So I checked the times and patches and after some confusion I do think it is correct. 

1. Regarding the times, the time indices in self.idxs correspond to the times in self.overlap_times (or the valid_times in self.tigge or self.mrms)
2. I illustrated the patches in the figure below. We have to be careful with the latitudes though, because the indices start at the highest lat-values and go to the lowest, following the standard in image processing (I think). (I was confused, but I think it's all consistent). I also was confused for a moment about the displayed rectangles below and whether the different patches overlap. But it is the red rectangles that include the last index, while when slicing, the last index is not actually included. (array[a:b] goes from a to b-1, not b itself.) So there is no overlap, just as we want it. 

![image](https://user-images.githubusercontent.com/43262391/103919179-0b8c4200-5110-11eb-9b03-e5f18bea18ce.png)

3. With padding, I think only the padding overlaps, which is probably what we want. 
4. I thoroughly checked the above points for the Tigge Data. After checking it for mrms, it also seems to be correct there.
